services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/migration/MigrationManagerImpl.java

```diff
@@ -526,19 +587,21 @@ public RestoreTypeResponse restoreRequest(UserInfo user, RestoreTypeRequest requ
 	 * @return
 	 */
 	@Override
-	public RestoreTypeResponse restoreStream(InputStream input, MigrationType primaryType,
-			BackupAliasType backupAliasType, long batchSize) {
+	public RestoreTypeResponse restoreStream(InputStream input, BackupManifest manifest) {
 		RestoreTypeResponse response = new RestoreTypeResponse();
-		if(!this.migratableTableDao.isMigrationTypeRegistered(primaryType)) {
+		if(!this.migratableTableDao.isMigrationTypeRegistered(manifest.getOriginalRequest().getMigrationType())) {
```

Comment:
We might have an edge case: if we want to promote a secondary to primary, but get rid of the original primary this will lead to not migrating any of the secondary. We might be able to avoid this doing this check in the loop below.